### ISSUE TITLE
pretrained resnet34 on tiny-imagenet

### ISSUE THREAD

**Issue by Sharpiless** [lehduong/GINP#2]
Thanks for your work. But I failed to train a resnet34 with 68.81% accuracy (54% actually). Could you please upload your pretrained model of the code for train from scratch?

**Comment by lehduong** [comment]
Hi Sharpiless. Thanks for your interest in our work. We used pre-trained models on Imagenet and fine-tune them on Tiny-Imagenet (modifying the pooling + fc layer). Did you use the same protocol?

**Comment by Sharpiless** [comment]
Thanks for your apply! That's the settings I used for training the resnet34:

```python
parser.add_argument('--data_root', default='data')
parser.add_argument('--model', default='resnet34_imagenet')
parser.add_argument('--dataset', default='tiny_imagenet')
parser.add_argument('--epochs', default=100, type=int, metavar='N',
                    help='number of total epochs to run')
parser.add_argument('-b', '--batch-size', default=128, type=int,
                    metavar='N',
                    help='mini-batch size (default: 256), this is the total '
                         'batch size of all GPUs on the current node when '
                         'using Data Parallel or Distributed Data Parallel')
parser.add_argument('--warm_up_epoches', default=10, type=int,
                    metavar='WPI', help='warm up epoches')
parser.add_argument('--warm_up_lr', default=0.01, type=int,
                    metavar='WPI', help='warm up learning rate')
parser.add_argument('--lr', '--learning-rate', default=0.1, type=float,
                    metavar='LR', help='initial learning rate', dest='lr')
parser.add_argument('--lr_decay_milestones', default="50,75", type=str,
                    help='milestones for learning rate decay')
parser.add_argument('--evaluate_only', action='store_true',
                    help='evaluate model on validation set')
parser.add_argument('--resume', default='', type=str, metavar='PATH',
                    help='path to latest checkpoint (default: none)')
parser.add_argument('--gpu', default=0, type=int,
                    help='GPU id to use.')
```

with pretrained weights on Imagenet and arguments with random crop and random flip. Could you give me some advice? Thanks a lot.

**Comment by Sharpiless** [comment]
Training code is modified from [https://github.com/zju-vipa/DataFree/blob/main/train_scratch.py](https://github.com/zju-vipa/DataFree/blob/main/train_scratch.py)

**Comment by lehduong** [comment]
Hi, I'm finding my old training scripts for tiny-imagenet. I'll try to reproduce the result and send it as soon as possible.

**Comment by Sharpiless** [comment]
Thank you very much for your timely reply and solving the problem!

**Comment by lehduong** [comment]
Hi @Sharpiless, please check out this colab for full training scripts: https://colab.research.google.com/gist/lehduong/fba7e14367e2b5aedd8315a6518203b1/tiny_imagenet_train.ipynb
